src/oc_erchef/apps/chef_index/src/chef_elasticsearch.erl

```diff
@@ -80,20 +112,49 @@ query_string_query_ejson(QueryString) ->
          }
     }.
 
+%%
+%% A note on deleting
+%%
+%% The delete-by-query API was removed from Elasticsearch in 2.0, and is only
+%% available as a plugin. As a result, we can't rely on it being there in external
+%% Elasticsearch instances we don't control. Rather, we replicate the delete-by-query
+%% by performing a search, iterating over the results, and then deleting the
+%% documents individually.
+%%
+%% Because the number of documents is unknown, we use the Scroll API, which
+%% allows us to return all the results without having to set size to some ungodly
+%% high number. While this does increase network traffic to/from Elasticsearch, it
+%% reduces the memory load. If network performance becomes a problem, we could
+%% increase the number of rows we return with each query.
 -spec delete_search_db_by_type(OrgId :: binary(), Type :: atom()) -> ok.
 delete_search_db_by_type(OrgId, Type)
   when Type == client orelse Type == data_bag_item orelse
        Type == environment orelse Type == node orelse
        Type == role ->
-    DeleteQuery = jiffy:encode({[query_string_query_ejson(
-                                   chef_index_query:search_db_from_orgid(OrgId) ++
-                                       "AND" ++
-                                       chef_index_query:search_type_constraint(Type))]}),
-    ok = chef_index_http:delete("/chef/_query", DeleteQuery).
+    Query = #chef_solr_query{
+                             rows = 1000,
+                             start = 0,
+                             search_provider = elasticsearch,
+                             query_string = chef_index_query:search_db_from_orgid(OrgId) ++
+                                "AND" ++ chef_index_query:search_type_constraint(Type)
+                            },
+    {ok, _, _, Ids} = search(Query),
+    delete_ids(Ids).
 
+-spec delete_search_db(OrgId :: binary()) -> ok.
 delete_search_db(OrgId) ->
-    DeleteQuery = jiffy:encode({[query_string_query_ejson(
-                                   chef_index_query:search_db_from_orgid(OrgId))]}),
-    ok = chef_index_http:delete("/chef/_query", DeleteQuery),
+    Query = #chef_solr_query{
+                             rows = 1000,
+                             start = 0,
+                             search_provider = elasticsearch,
+                             query_string = chef_index_query:search_db_from_orgid(OrgId)
+                            },
+    {ok, _, _, Ids} = search(Query),
+    delete_ids(Ids).
+
+delete_ids([]) ->
     ok = commit(),
-    ok.
+    ok;
+delete_ids([Id | Ids]) ->
+    ok = chef_index_http:delete("/chef/object/" ++ Id, []),
```

Comment:
That was not the intent, no. We should confirm with @charlesjohnson if thats the behavior he wants when we are re-indexing.